prediction_market_agent/agents/autogen_general_agent/farcaster_agent.py

```diff
@@ -0,0 +1,148 @@
+import datetime
+from enum import Enum
+from string import Template
+from typing import Any, Dict, Optional
+
+import autogen
+from autogen import AssistantAgent, UserProxyAgent
+from autogen.cache import Cache
+from loguru import logger
+from prediction_market_agent_tooling.markets.agent_market import FilterBy, SortBy
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
+    OmenSubgraphHandler,
+)
+from prediction_market_agent_tooling.tools.utils import utcnow
+
+from prediction_market_agent.agents.autogen_general_agent.prompts import (
+    INFLUENCER_PROMPT,
+)
+from prediction_market_agent.utils import APIKeys
+
+# We have an influencer and a critic. Based on https://microsoft.github.io/autogen/docs/notebooks/agentchat_nestedchat/.
+
+
+class AutogenAgentType(str, Enum):
+    WRITER = "writer"
+    CRITIC = "critic"
+    USER = "user"
+
+
+def reflection_message(
+    recipient: UserProxyAgent,
+    messages: list[Dict[Any, Any]],
+    sender: AssistantAgent,
+    config: Optional[Any] = None,
+) -> str:
+    print("Reflecting...", "yellow")
+    return f"""
+        Reflect and provide critique on the following tweet. \n\n {recipient.chat_messages_for_summary(sender)[-1]['content']}
+        Note that it should not include inappropriate language.
+        Note also that the tweet should not sound robotic, instead as human-like as possible.
+        Also make sure to ask the recipient for an improved version of the tweet, following your critic, and nothing else.
+        """
+
+
+def build_llm_config(model: str) -> Dict[str, Any]:
+    keys = APIKeys()
+    return {
+        "config_list": [
+            {"model": model, "api_key": keys.openai_api_key.get_secret_value()}
+        ],
+    }
+
+
+def build_agents(model: str) -> Dict[AutogenAgentType, autogen.ConversableAgent]:
```

Comment:
We have 3 agents, following the Autogen example:
-> User: handles user input, prompts, relays messages between writer <> critic
-> Writer: writes a tweet, receives feedback from critic, rewrites it
-> Critic: assesses tweet and gives suggestions